htlcswitch/link.go

```diff
@@ -37,9 +37,11 @@ const (
 	// willing to lock its own funds too, could force the funds of this node
 	// to be locked up for an indefinite (max int32) number of blocks.
 	//
-	// The value 5000 is based on the maximum number of hops (20), the
-	// default cltv delta (144) and some extra margin.
-	DefaultMaxOutgoingCltvExpiry = 5000
+	// The value 1008 corresponds to on average one week worth of blocks and
+	// is based on the maximum number of hops (20), the default cltv delta
+	// (40) and some extra margin to account for the other lightning
+	// implementations.
+	DefaultMaxOutgoingCltvExpiry = 1008
```

Comment:
IMO cutting this value by 80% is far too drastic, i'd prefer to slowly reduce this value over time to prevent any unforeseen interactions with pathfinding and other implementations.

fwiw, after the variable-size onion payload is deployed, the max pathlength will be closer to 26. the extra margin included in the original value was 5000 - 144 * 20 = 2120 blocks. Adding that to say 40*26 would give a value of 3160. Somewhere around 3k I think a more reasonable reduction. We can drop this again in the future if we don't see any regressions

Reply:
3k would still result in a _20 day_ timelock for HTLCs that will be auto accepted. That's even longer than our max CSV delay (where most of the funds ideally are in a channel). Those values are based on the max path length, which afaik based on the current graph diameter, is nowhere close to being hit. If there're nodes today that are routinely extending paths of length 20, then that likely indicates they're attempting to launch nuisance attacks on the network.

Reply:
all of that makes sense. i'm only expressing concern over how dramatic the drop si, not that we shouldn't shoot to end at this lower value.

Reply:
One could also argue that the previous value didn't need to be so high.